### PR TITLE
[MODULARIZING] Removes Foreigner and Adds Quirk Sanitization

### PR DESCRIPTION
**Pull request by RimiNosha**
## About The Pull Request
BEFORE YOU REACT WITH 👎, READ!!

We already have foreigner functionality with our current prefs system. This quirk is redundant, and introduces a bunch of weird behavior on our custom species, and honestly, is more effort than it's worth when we've had this underlying functionality since newprefs became a thing.

That, and I'd rather allow people to be foreigner with any language they want!

Also, you're able to start with a recorder at round start with the loadout system, so that's already covered.
image
image

## How This Contributes To The Skyrat Roleplay Experience
Less random language breakages, and allows for some more quirky characters (ironically) than the quirk allowed.

## Changelog

:cl:
add: Added quirk sanitization.
remove: Removed Foreigner. You are able to recreate this quirk almost exactly via removing common, and optionally giving yourself a recorder via loadout.
/:cl:

### CODE REVIEW

**Comment by desertmote** [comment]
foreigner quirk allowed more jobs to be played. consider opening some up, as you have it here it's very strict. e.g: miner

**Comment by Iamgoofball** [comment]
:-1: removing a vanilla /tg/ quirk rather than fixing the bugs created by skyrat's code is not a good idea, foreigner breaking means we have issues with our language code, not that foreigner itself is broken

this is removing the train because a stretch of the rail it drove over is broken

File: code/modules/client/preferences.dm [hunk 1]
```diff
@@ -113,8 +113,9 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 	var/loaded_preferences_successfully = load_preferences()
 	if(loaded_preferences_successfully)
 		if(load_character())
-			// SKYRAT EDIT START - Sanitizing languages
+			// SKYRAT EDIT START - Sanitizing preferences
 			sanitize_languages()
+			sanitize_quirks()
```
Comment (GoldenAlpharex) [inline code/modules/client/preferences.dm:118]: Good call.

File: modular_skyrat/modules/customization/modules/client/preferences.dm [hunk 2]
```diff
@@ -14,3 +14,19 @@
...
+		if(isnull(quirk))
+			all_quirks.Remove(quirk)
+			quirks_edited = TRUE
+			continue
+
+		quirk = new quirk()
+		// Explanation for this is above.
+		if(!(quirk.type in subtypesof(/datum/quirk)) || quirk.hidden_quirk)
+			all_quirks.Remove(quirk)
+			quirks_edited = TRUE
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/customization/modules/client/preferences.dm:31]:
```suggestion
			quirks_edited = TRUE

```

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]
We'll have to test-merge this first, but I think this is a good change.

**Review by GoldenAlpharex — APPROVED** [review]

**Pull request merged**